include/windows/pthread.h

```diff
@@ -41,7 +41,7 @@
 #include <stdlib.h>
 #include <stdbool.h>
 
-#define PTHREAD_MUTEX_INITIALIZER {0}
+#define PTHREAD_MUTEX_INITIALIZER {-1, -1, 0, 0, 0, 0}
```

Comment:
The problem with creating an error macro is that it forces non-Windows builds to have an initialization routine somewhere else as well.  But I think it's worth looking at, since it would catch these problems.

Which lock caused the issue?  The ini and util_fabric_lock are initialized in windows specific code.  I would have expected the locks in mem_monitor to be linux only, but may there's generic code there that uses them.